src/verification.rs

```diff
@@ -480,9 +479,8 @@ pub trait NonceVerifier {
 impl NonceVerifier for &Nonce {
     fn verify(self, nonce: Option<&Nonce>) -> Result<(), String> {
         if let Some(claims_nonce) = nonce {
-            if verify_slices_are_equal(claims_nonce.secret().as_bytes(), self.secret().as_bytes())
-                .is_err()
-            {
+            // Nonce::eq is already implemented with a constant time comparison
+            if !claims_nonce.eq(self) {
```

Comment:
does `claims_nonce != &self` (or similar) work here?

Reply:
True, I guess that would be the same.

I guess I wrote too much Java code lately :D